klayout/drc/rule_decks/hres.drc

```diff
@@ -0,0 +1,150 @@
+  hres_poly = poly2.interacting(pplus).interacting(sab).interacting(res_mk).interacting(resistor)
+  hres1_poly = poly2.interacting(pplus).interacting(sab).interacting(res_mk)
+
+  # Rule HRES.1: Minimum space. Note : Merge if the spacing is less than 0.4 um. is 0.4µm
+  logger.info('Executing rule HRES.1')
+  hres1_l1 = resistor.interacting(hres1_poly).space(0.4.um, euclidian)
+  hres1_l1.output('HRES.1', 'HRES.1 : Minimum space. Note : Merge if the spacing is less than 0.4 um. : 0.4µm')
+  hres1_l1.forget
+
+  # Rule HRES.2: Minimum width of Poly2 resistor. is 1µm
+  logger.info('Executing rule HRES.2')
+  hres2_l1 = hres_poly.width(1.um, euclidian)
+  hres2_l1.output('HRES.2', 'HRES.2 : Minimum width of Poly2 resistor. : 1µm')
+  hres2_l1.forget
+
+  # Rule HRES.3: Minimum space between Poly2 resistors. is 0.4µm
+  logger.info('Executing rule HRES.3')
+  hres3_l1 = hres_poly.space(0.4.um, euclidian)
+  hres3_l1.output('HRES.3', 'HRES.3 : Minimum space between Poly2 resistors. : 0.4µm')
+  hres3_l1.forget
+
+  # Rule HRES.4: Minimum RESISTOR overlap of Poly2 resistor. is 0.4µm
+  logger.info('Executing rule HRES.4')
+  hres4_l1 = hres_poly.enclosed(resistor, 0.4.um, euclidian).polygons(0.001.um)
+  hres4_l2 = hres_poly.not_outside(resistor).not(resistor)
+  hres4_l  = hres4_l1.or(hres4_l2)
+  hres4_l.output('HRES.4', 'HRES.4 : Minimum RESISTOR overlap of Poly2 resistor. : 0.4µm')
+  hres4_l1.forget
+  hres4_l2.forget
+  hres4_l.forget
+
+  # Rule HRES.5: Minimum RESISTOR space to unrelated Poly2. is 0.3µm
+  logger.info('Executing rule HRES.5')
+  hres5_l1 = resistor.interacting(hres1_poly).separation(poly2.not_interacting(sab), 0.3.um, euclidian)
+  hres5_l1.output('HRES.5', 'HRES.5 : Minimum RESISTOR space to unrelated Poly2. : 0.3µm')
+  hres5_l1.forget
+
+  # Rule HRES.6: Minimum RESISTOR space to COMP.
+  logger.info('Executing rule HRES.6')
+  hres6_l1 = resistor.interacting(hres1_poly).separation(comp, 0.3.um, euclidian).polygons(0.001.um)
+  hres6_l2 = comp.not_outside(resistor.interacting(hres1_poly))
+  hres6_l = hres6_l1.or(hres6_l2)
+  hres6_l.output('HRES.6', 'HRES.6 : Minimum RESISTOR space to COMP.')
+  hres6_l.forget
+  hres6_l1.forget
+  hres6_l2.forget
+  hres1_poly.forget
+
+  # Rule HRES.7: Minimum Pplus overlap of contact on Poly2 resistor. is 0.2µm
+  logger.info('Executing rule HRES.7')
+  hres7_l1 = contact.and(hres_poly).enclosed(pplus, 0.2.um, euclidian).polygons(0.001.um)
+  hres7_l2 = contact.and(hres_poly).not_outside(pplus).not(pplus)
+  hres7_l  = hres7_l1.or(hres7_l2)
+  hres7_l.output('HRES.7', 'HRES.7 : Minimum Pplus overlap of contact on Poly2 resistor. : 0.2µm')
+  hres7_l1.forget
+  hres7_l2.forget
+  hres7_l.forget
+
+  # Rule HRES.8: Space from salicide block to contact on Poly2 resistor.
+  logger.info('Executing rule HRES.8')
+  hres8_l1 = contact.and(hres_poly).separation(sab,
+                                               0.22.um).polygons(0.001.um).or(contact.and(hres_poly).interacting(sab))
```

Comment:
why do we have `.001` here?

Reply:
can we use `1.dbu` instead?

Reply:
@proppy Same comment.

Reply:
Is that a workaround for a KLayout bug?

Reply:
according to https://github.com/google/globalfoundries-pdk-libs-gf180mcu_fd_pv/pull/65#discussion_r1158437563 this is to convert edges to polygons, I suggested using `simple_merge_e2p` instead but @atorkmabrains is not sure if this is available in the DRC context.

Reply:
filed #66 to investigate this further